Source/WebCore/page/EventHandler.cpp

```diff
@@ -1649,9 +1649,13 @@ std::optional<Cursor> EventHandler::selectCursor(const HitTestResult& result, bo
             return handCursor();
 
         bool inResizer = false;
-        if (renderer && renderer->hasLayer()) {
-            // FIXME: With right-aligned text in a box, the renderer here is usually a RenderText, which prevents showing the resize cursor: webkit.org/b/210935.
-            auto& layerRenderer = downcast<RenderLayerModelObject>(*renderer);
+        auto resizerRenderer = renderer;
+
+        if (resizerRenderer && dynamicDowncast<RenderText>(*resizerRenderer) && resizerRenderer->parent()->hasLayer())
```

Comment:
You could use `is<RenderText>(resizerRenderer)` to both do the null check and the type check.

Is the `resizerRenderer->parent()->hasLayer())` check doing anything useful here?